src/execution_plan/ops/op_aggregate.c

```diff
@@ -173,7 +173,7 @@ static Record _handoff(OpAggregate *op) {
 		int rec_idx = op->record_offsets[i + op->key_count];
 		AR_ExpNode *exp = group->aggregationFunctions[i];
 		AR_EXP_Reduce(exp);
-		SIValue res = AR_EXP_Evaluate(exp, NULL);
+		SIValue res = AR_EXP_Evaluate(exp, r);
```

Comment:
It's not, this handles the case where the list expression is an aggregation function:
`MATCH p=()-[*]->() RETURN [x IN collect(p)]`

I'm actually somewhat concerned about more complicated variants of this case, because the list comprehension should be considered part of the group key rather than the group's aggregation functions, but we currently differentiate them just using `AR_EXP_ContainsAggregation`. I fear that we may actually need to use separate Aggregate and Project ops for some cases (related to #1057), but need to give it more thought.